### PR TITLE
chore: improve build and CI/CD 

### LINKED ISSUE #3: CI/CD
## TODO:
- [x] build scripts that generate optimized svg, jpg, png, webp, and eps artifacts, including red and white variants
- [x] ci/cd which adds artifacts to github releases
- [x] documentation, readme, contributing.md, etc
- [x] ~~automation to detect name conflicts (git hooks?)~~: Not doing this right now, can revisit later if deemed an issue
- [x] linting/verifying monotone icons work properly (no `fill` or `fill="currentColor"`)

### PR DESCRIPTION
**Pull request by zeroedin**
## What I did

1. Add wireit dependency to manage scripts
2. Add changesets for release flow
3. Add github workflow for release
4. Improved build, adding license comments and outputting both `js` and `svg` into `dist`

Closes #3 

### TODO
- [x] ~~Determine if the [preexisting changeset ](https://github.com/RedHat-UX/red-hat-icons/blob/chore/improve-build-ci-cd/.changeset/khaki-avocados-sort.md) should exist still~~: No package was released prior, rework changeset for 1.0.0
- [x] ~~Determine if we are going to release a beta/alpha first to ensure proper release~~: We are going straight to 1.0.0
- [x] Improved README & Contributing documentation - [waiting on feedback about contributing from brand]
- [x] Setup netlify to publish documentation, improve 11ty documentation page
- [x] Close remaining issues https://github.com/RedHat-UX/red-hat-icons/issues/3

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  Your svg files will then be compressed and optimized, new artifacts will be generated in `./dist`
+3. Check your work in the demo
+  ```bash
+  npm run serve
+  ```
+4. [Commit](#commit-changes) your changes.
+
+## Commit Changes 
+1. Commit your changes using [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) style.
+2. Open a new PR and request team members for review.
```
Comment (bennypowers) [inline CONTRIBUTING.md:50]: Add changeset

File: LICENSE
Comment (bennypowers) [inline LICENSE:1]: Replace with mit

File: README.md [hunk 2]
```diff
@@ -1,2 +1,119 @@
 # Red Hat Icons
 Official Red Hat Brand Icons
+
+
+## Licensing
+
```
Comment (bennypowers) [inline README.md:6]: Maybe we should check of the JS modules should stay MIT, but retain the license comment in the exported svg  node

File: README.md [hunk 3]
```diff
@@ -1,2 +1,119 @@
...
+The software contained in this repository as well as the artifacts contained and produced are covered by two separate licenses. A [MIT license](./LICENSE) covers the build scripts and configuration, as well as a [Creative Commons 4.0 License](#creative-commons-40) which covers the source and build icon artifacts including the SVG icons and JS module variants.
+
+
+### Creative Commons 4.0
```
Comment (bennypowers) [inline README.md:10]: Instead of pasting this in README, move to LICENSE_ICONS.md

**Review by bennypowers — CHANGES_REQUESTED** [review]

File: LICENSE
Comment (zeroedin) [reply LICENSE:1]: Pretty sure it is MIT? https://opensource.org/license/MIT

File: README.md (hunk 2 above)
Comment (zeroedin) [reply README.md:6]: Reworded the README a bit for this section, let me know if you feel that better applies.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+## Documentation
+To work on this project's documentation first install the project on your local machine:
+
+1. [Install Node Version Manager](Install Node Version Manager) for your system.
+2. Navigate to your project folder
+  ```bash
+  cd red-hat-icons
+  ```
+3. Run [Node Version Manager's](https://github.com/nvm-sh/nvm) `use` command to switch to the projects [Node](https://nodejs.org/en) version.
```
Comment (adamjohnson) [inline CONTRIBUTING.md:15]: I think this NVM link was meant to be linked in step #1.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+  ```bash
+  npm run serve
+  ```
+4. [Commit](#commit-changes) your changes.
+
+## Commit Changes 
+1. Commit your changes using [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) style.
+2. Add a [changeset](https://github.com/changesets/changesets) 
+  ```bash
+  npx changeset
```
Comment (adamjohnson) [inline CONTRIBUTING.md:52]: New contributors are unlikely to know how changeset works. Might be worthwhile to add some more details here ("what should I select when it prompts me?").

File: README.md [hunk 6]
```diff
@@ -1,2 +1,7 @@
 # Red Hat Icons
 Official Red Hat Brand Icons
+
```
Comment (adamjohnson) [inline README.md:3]: Add:

  * Link to https://red-hat-icons.netlify.app/
  * How to use / "Quick Start"
    * Copy -> Paste or `npm i` into your project or ??
  * Link to contributing.md "# Contribute, Looking to contribute, see our [contribution instructions](#)."
  * Link to other icons / icon libraries used by Red Hat (if applicable).

File: index.html [hunk 7]
```diff
@@ -84,22 +149,52 @@
...
         {% for icon in set.icons %}
         <figure>
           {% include set.dir + '/' + icon + '.svg' %}
           <figcaption>
-            <rh-tooltip>
+            {{ icon }}
+             <rh-tooltip position="bottom">
+              <rh-button variant="link" data-html='<rh-icon set="{{ set.set }}" icon="{{ icon }}"></rh-icon>'>
+                <pf-icon set="far" icon="copy" size="md" loading="eager"></pf-icon>
+                <span class="sr-only">Click to</span> Copy <span class="sr-only">rh-icon {{icon}} from set {{set}}
```
Comment (adamjohnson) [inline index.html:194]: Currently prints "rh-icon caret-right from set [object Object]"

File: index.html [hunk 8]
```diff
@@ -84,22 +149,52 @@
...
+        })
+      })
+    </script>    
   </head>
   <body>
     <header><h1>Red Hat Icons</h1></header>
     <main>{% for set in iconSets %}
-      <section class="{{ set.set }}">
-        <h2>{{ set.set }}</h2>
+      {% set id = set.set | lower %}
```
Comment (adamjohnson) [inline index.html:176]: Add accessible toast for "Copied" after clicking the copy button. Probably display this in the upper right corner of the screen.

Don't forget to add `role="status"`, eg:

```html
<div class="toast" role="status">
  <p>Copied!</p>
</div>
```

**Review by adamjohnson — CHANGES_REQUESTED** [review]

**Review by markcaron — APPROVED** [review]
Liked Greatly; Toggle-button Mashed!

File: docs/index.html [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+        <h2 id="{{ id }}">{{ section }}</h2>
+        {% for icon in set.icons %}
+        <figure>
+          {% include set.dir + '/' + icon + '.svg' %}
+          <figcaption>
+            {{ icon }}
+             <rh-tooltip position="bottom">
+              <rh-button name="toast" variant="link" data-html='<rh-icon set="{{ set.set }}" icon="{{ icon }}"></rh-icon>' data-icon="{{icon}}">
+                <pf-icon set="far" icon="copy" size="md" loading="eager"></pf-icon>
+                <span class="sr-only">Click to</span> Copy <span class="sr-only">rh-icon {{icon}} from set {{set.set}}
```
Comment (adamjohnson) [inline docs/index.html:86]: This is [**very** verbose](https://github.com/user-attachments/assets/224741a7-4584-43a4-b8c9-9506a9b1e087). Right now, it's:

> Click to copy rh-icon arrow-down from set microns

...repeated 100's of times.

Consider changing it to:

"Copy arrow-down"

or:

"Copy arrow down markup"

("markup" could be "code", "icon", etc)

File: docs/index.html [hunk 10]
```diff
@@ -0,0 +1,103 @@
...
+          const { html, icon } = event.target.dataset;
+          console.log(event.target.dataset);
+          copy(html); 
+          await Toast(`Copied ${icon}`);
+        });
+      })
+    </script>    
+  </head>
+  <body>
+    <header><h1>Red Hat Icons</h1></header>
```
Comment (adamjohnson) [inline docs/index.html:65]: You may consider adding a jump-link nav under the h1 to each icon set. YMMV.

Comment (adamjohnson) [reply docs/index.html:65]: If you do this, you may also consider adding a skip link.

File: README.md [hunk 11]
```diff
@@ -1,2 +1,22 @@
...
+```bash
+npm install @rhds/icons
+```
+
+## Contributing
+
+Please read the [contributing guide](./CONTRIBUTING.md) for more information.
+
+## Demo
```
Comment (adamjohnson) [inline README.md:15]: Consider moving this under the h1. People probably just wanna see the demo!

File: README.md [hunk 12]
```diff
@@ -1,2 +1,22 @@
 # Red Hat Icons
+
 Official Red Hat Brand Icons
+
+## Quickstart
+
+```bash
+npm install @rhds/icons
+```
+
```
Comment (adamjohnson) [inline README.md:10]: > 2. Visit the [demo](https://deploy-preview-6--red-hat-icons.netlify.app/). 3. Copy your desired icon. 4. Paste it into your project.

**Review by adamjohnson — CHANGES_REQUESTED** [review]

**Review by adamjohnson — APPROVED** [review]
Logical Graphics, Thorough Manuals 👏

**Pull request merged**